microcosm_postgres/operations.py

```diff
@@ -65,13 +66,37 @@ def drop_alembic_table(graph):
         return True
 
 
+# Cached database metadata instance
+_metadata = None
+
+
 def recreate_all(graph):
     """
-    Drop and add back all database tables.
+    Drop and add back all database tables, or reset all data associated with a database.
+    Intended mainly for testing, where a test database may either need to be re-initialized
+    or cleared out between tests
 
     """
-    drop_all(graph)
-    create_all(graph)
+
+    global _metadata
+    initialize = False
+
+    if _metadata is None:
+        _metadata = MetaData(bind=graph.postgres, reflect=True)
+        initialize = True
+
+    # First use of this function, re-create the database from scratch.
+    if initialize:
+        drop_all(graph)
```

Comment:
I was initially playing with passing in the same metadata instance here as well, but I was running into very odd issues (mostly revolving around not being able to drop constraints). If desired I could probably find a way, but this works.